airflow/jobs.py

```diff
@@ -408,6 +409,21 @@ def process_dag(self, dag, executor):
                 ti = ti_dict[task.task_id]
                 ti.task = task  # Hacky but worky
                 if ti.state == State.RUNNING:
+                    if ti.key not in executor.running:
```

Comment:
I don't think this is compatible with the way we currently run the scheduler at Airbnb. Also not compatible with a future where we'd run multiple instances of the scheduler... I think ultimately we want the database to be the source of truth as far as status goes, but use some of the signals from the executor to force double checking things or alert about unusual events.